lib/circuitbox/circuit_breaker.rb

```diff
@@ -130,19 +130,26 @@ def try_close_next_time
     private
 
     def should_open?
-      failures = failure_count
-      successes = success_count
-      rate = error_rate(failures, successes)
-
-      passed_volume_threshold?(failures, successes) && passed_rate_threshold?(rate)
+      aligned_time = align_time_to_window
+
+      failures, successes = @circuit_store.values_at(stat_storage_key('failure', aligned_time),
+                                                     stat_storage_key('success', aligned_time),
+                                                     raw: true)
+      # Calling to_i is only needed for moneta stores which can return a string representation of an integer.
+      # While readability could increase by adding .map(&:to_i) to the end of the values_at call it's also slightly
+      # less performant when we only have two values to convert.
```

Comment:
> `# While readability could increase by adding .map(&:to_i) to the end of the values_at call it's also slightly`
> `# less performant when we only have two values to convert.`

since we're using `values_at` in this one place only, I'd also recommend not to do an implicit array conversion using the splat operator as well. We might as well just expect 2 keys in the underlying method - 
```ruby
    def values_at(success_key, failure_key)
      @mutex.synchronize do
        current_time = current_second
        success = fetch_container(success_key, current_time)&.value
        failure = fetch_container(failure_key, current_time)&.value
        return success, failure
      end
    end
```
I also noticed we aren't actually using the `_opts` arg in both `load` and `values_at` method. Can we remove them from the definition as well?

Reply:
> I also noticed we aren't actually using the `_opts` arg in both `load` and `values_at` method. Can we remove them from the definition as well?

The reason both `load` and `values_at` support options but don't use them is options is required for moneta stores and `Circuitbox::MemoryStore` follows the same interface as moneta. For some of the keys in `Circuitbox::CircuitBreaker` the option `raw: true` is passed to support features like moneta's [atomic increment](https://github.com/moneta-rb/moneta/tree/v1.6.0#atomic-incrementation-and-raw-access).

> > `# While readability could increase by adding .map(&:to_i) to the end of the values_at call it's also slightly`
> > `# less performant when we only have two values to convert.`
> 
> since we're using `values_at` in this one place only, I'd also recommend not to do an implicit array conversion using the splat operator as well. We might as well just expect 2 keys in the underlying method -
> 
> ```ruby
>     def values_at(success_key, failure_key)
>       @mutex.synchronize do
>         current_time = current_second
>         success = fetch_container(success_key, current_time)&.value
>         failure = fetch_container(failure_key, current_time)&.value
>         return success, failure
>       end
>     end
> ```

Similar to the options question, `Circuitbox::MemoryStore` is following the [same interface](https://github.com/moneta-rb/moneta/tree/v1.6.0#moneta-api) as moneta which is why `values_at` is defined the way it is. While it's currently possible to define `values_at` like `def values_at(failure_key, success_key, _opts = {})` it causes the interfaces between `Circuitbox::MemoryStore` and moneta to diverge.